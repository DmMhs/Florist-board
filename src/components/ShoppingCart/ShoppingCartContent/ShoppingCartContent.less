.fix {
  border: none;
  border-radius: 5px;
  padding: 9px;
  font-size: 2rem;
  color: #333333;
  background-color: #eee;
  box-shadow: 1px 1px 5px #ccc;
  margin-bottom: 2rem;
  span {
    font-weight: lighter;
  }
  &:hover {
    cursor: pointer;
    background-color: rgb(181, 235, 181);
  }
}
