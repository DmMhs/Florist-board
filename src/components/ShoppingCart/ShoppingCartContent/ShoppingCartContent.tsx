import React, { useContext } from 'react';

import './ShoppingCartContent.less';
import { AuthContext } from '../../Auth/AuthContext';
import labels from '../../../config/labels';

interface ShoppingCartContentProps {
  totalPrice: number;
  cartItemsList: JSX.Element[];
  fixEmptyCart:
    | ((event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void)
    | undefined;
}

const ShoppingCartContent = (props: ShoppingCartContentProps) => {
  const { totalPrice, cartItemsList } = props;
  const context = useContext(AuthContext);
  const totalPriceFormatted = totalPrice.toFixed(2);
  return cartItemsList.length > 0 ? (
    <div>
      <ul>
        <li>
          <span>
            {labels[context.state.lang as string].pages.shop.cart.title}
          </span>
          <span />
          <span>
            {labels[context.state.lang as string].pages.shop.cart.amount}
          </span>
          <span>
            {labels[context.state.lang as string].pages.shop.cart.price}
          </span>
          <span />
        </li>
        <hr />
        {cartItemsList}
      </ul>
      <div className="order">
        <p>
          {labels[context.state.lang as string].pages.shop.cart.total}{' '}
          {totalPriceFormatted}$
        </p>
        <button type="button" className="order-btn">
          {labels[context.state.lang as string].pages.shop.cart.orderBtn}
        </button>
      </div>
    </div>
  ) : (
    <div>
      <h4>{labels[context.state.lang as string].pages.shop.cart.empty}</h4>
      <button type="button" className="fix" onClick={props.fixEmptyCart}>
        <span>{labels[context.state.lang as string].pages.shop.cart.btn}</span>{' '}
        <i className="far fa-smile-wink" />
      </button>
    </div>
  );
};

export default ShoppingCartContent;
