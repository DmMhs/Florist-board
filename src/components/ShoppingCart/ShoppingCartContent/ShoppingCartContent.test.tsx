import React from 'react';
import { shallow } from 'enzyme';

import ShoppingCartContent from './ShoppingCartContent';

it('matches a snapshot', () => {
  const TestJSX = <span key={1}>jast a test</span>;
  const wrapper = shallow(
    <ShoppingCartContent
      totalPrice={100}
      cartItemsList={[TestJSX, TestJSX]}
      fixEmptyCart={() => {}}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
