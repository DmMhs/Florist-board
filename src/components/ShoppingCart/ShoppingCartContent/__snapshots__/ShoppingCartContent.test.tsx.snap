// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshot 1`] = `
<div>
  <ul>
    <li>
      <span>
        PRODUCT
      </span>
      <span />
      <span>
        AMOUNT
      </span>
      <span>
        PRICE
      </span>
      <span />
    </li>
    <hr />
    <span
      key="1"
    >
      jast a test
    </span>
    <span
      key="1"
    >
      jast a test
    </span>
  </ul>
  <div
    className="order"
  >
    <p>
      Total Sum: 
      100.00
      $
    </p>
    <button
      className="order-btn"
      type="button"
    >
      ORDER
    </button>
  </div>
</div>
`;
